### PR TITLE
Add recipes rest api

### PR DESCRIPTION
**Pull request by gsouzab**
The recipes API for the [onboarding exercise](https://www.notion.so/travelperk/Description-of-the-exercise-5db39976c0b34ff0a10ed5c84a6f7fe9). Mostly using resources from the [DRF website](https://www.django-rest-framework.org/) (tutorials, quickstart...) and the udemy course ([this repo](https://github.com/LondonAppDeveloper/c2-recipe-app-api-2)).

Testing only API (integration)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,27 @@
 # Recipe onboarding app
+CRUD API with Django and DRF for the onboarding.
+
+## Starting the app
+```shell
+docker compose up
+```
```
Comment (danielsto) [inline README.md:7]: Nice readme btw ❤️ 

I think it's just missing the migration, at least it was necessary for me to run the app
```shell
docker compose run --rm app sh -c "python manage.py migrate"
```

File: requirements.txt [hunk 2]
```diff
@@ -1 +1,3 @@
 Django>=4.0,<5.0
+psycopg2>=2.9.5,<3.0
+djangorestframework>=3.14.0,<4.0
```
Comment (danielsto) [inline requirements.txt]: It's a good idea to list the exact versions of the packages in the `requirements.txt` file. It's easier to recreate the environment and avoid compatibility issues 👍🏼

File: app/apps/recipes/serializers.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  ingredients = IngredientsSerializer(many=True, allow_empty=False)
+  class Meta:
+    model = Recipe
+    fields = ['id', 'name', 'description', 'ingredients']
+
+  def create(self, validated_data):
+    ingredients_data = validated_data.pop('ingredients')
+    recipe = Recipe.objects.create(**validated_data)
+    for ingredient_data in ingredients_data:
+      Ingredient.objects.create(recipe=recipe, **ingredient_data)
```
Comment (danielsto) [inline app/apps/recipes/serializers.py]: Suggestion: when creating several objects, Django has a standarized and efficient way of doing it with [`bulk_create`](https://docs.djangoproject.com/en/dev/ref/models/querysets/#bulk-create).

Comment (danielsto) [reply app/apps/recipes/serializers.py]:
```python
def create(self, validated_data):
    ingredients_data = validated_data.pop('ingredients')
    recipe = Recipe.objects.create(**validated_data)

    ingredients = []
    for ingredient_data in ingredients_data:
        ingredient = Ingredient(recipe=recipe, **ingredient_data)
        ingredients.append(ingredient)

    Ingredient.objects.bulk_create(ingredients)
    return recipe
```

Something like that

File: app/apps/recipes/models.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+from django.db import models
+
+class Recipe(models.Model):
+  """Recipe model"""
+  name = models.CharField(max_length=255)
+  description = models.TextField()
+
+class Ingredient(models.Model):
+  """Ingredient model"""
+  name = models.CharField(max_length=255)
```
Comment (danielsto) [inline app/apps/recipes/models.py]: I'd avoid setting a character limit if theres no specific reason for it. Also there's no real advantage at database level with Postgres ([docs](https://www.postgresql.org/docs/14/datatype-character.html)).

Comment (chrbarreto-travelperk) [reply app/apps/recipes/models.py]: We avoid using `CharField` because it has no advantage with postgres. `TextField` is the recommended way to go.

Comment (gsouzab) [reply app/apps/recipes/models.py]: Nice, didn't know about that.

File: README.md (hunk 1 above)
Comment (gsouzab) [reply README.md:7]: Updated the up command to run the migrations as well.

**Review by danielsto — APPROVED** [review]
I think it looks good 👍🏼

**Comment by facundolinlaud** [comment]
Took a look yesterday and LGTM as well!

**Pull request merged**